src/EventStore.Core/Services/Transport/Http/Authentication/NodeCertificateAuthenticationProvider.cs

```diff
@@ -90,24 +90,32 @@ public class NodeCertificateAuthenticationProvider : IHttpAuthenticationProvider
 				return false;
 			}
 
+			bool hasIpOrDnsSan = clientCertificate.GetSubjectAlternativeNames()
+				.Where(x => x.type is CertificateNameType.DnsName or CertificateNameType.IpAddress)
+				.IsNotEmpty();
+
+			if (!isServerCertificate && !hasReservedNodeCN && !hasIpOrDnsSan) {
+				// We are sure that this is a client certificate,
+				// Not a misconfigured server certificate with incorrect EKUs
```

Comment:
```suggestion
				// We are sure that this is not a misconfigured node certificate with incorrect EKUs, missing SANs, etc. It could be a user certificate.
```